Sources/ArcGISToolkit/Components/Popups/MediaPopupElementView.swift

```diff
@@ -82,8 +82,10 @@ struct MediaPopupElementView: View {
                     }
                 }
             }
-            .onSizeChange {
-                width = $0.width * widthScaleFactor
+            .onGeometryChange(for: CGRect.self) { proxy in
+                proxy.frame(in: .global)
+            } action: { newValue in
+                width = newValue.width * widthScaleFactor
             }
```

Comment:
```suggestion
            .onGeometryChange(for: CGFloat.self) { proxy in
                proxy.frame(in: .global).width
            } action: { newValue in
                width = newValue * widthScaleFactor
            }
```

Again, consider using the newer API when available:

```swift
let scrollView = ...
if #available(iOS 18.0, *) {
    scrollView
        .onScrollGeometryChange(for: CGFloat.self) { geometry in
            geometry.contentSize.width
        } action: { _, newValue in
            width = newValue * widthScaleFactor
        }

} else {
    scrollView
        .onGeometryChange(for: CGFloat.self) { proxy in
            proxy.frame(in: .global).width
        } action: { newValue in
            width = newValue * widthScaleFactor
        }
}
```

Reply:
Did you have this working on Catalyst? I'm hitting an error and rendering a blank view:

<p align="center">
  <img width="50%" src="https://github.com/user-attachments/assets/124e3e0c-5592-446d-84c4-f6920b8084a8">
</p>

<details>

<summary>Swift Code</summary>

```swift
        var body: some View {
            let scrollView = ScrollView(.horizontal) {
                HStack(alignment: .top, spacing: 8) {
                    ForEach(popupMedia) { media in
                        Group {
                            switch media.kind {
                            case .image:
                                ImageMediaView(
                                    popupMedia: media,
                                    mediaSize: mediaSize
                                )
                            case .barChart, .columnChart, .lineChart, .pieChart:
                                ChartMediaView(
                                    popupMedia: media,
                                    mediaSize: mediaSize
                                )
                            default:
                                EmptyView()
                            }
                        }
                        .frame(width: mediaSize.width, height: mediaSize.height)
                        .contentShape(RoundedRectangle(cornerRadius: 8))
                    }
                }
            }
            if #available(iOS 18.0, *) {
                scrollView
                    .onScrollGeometryChange(for: CGFloat.self) { geometry in
                        geometry.contentSize.width
                    } action: { _, newValue in
                        width = newValue * widthScaleFactor
                    }
            } else {
                scrollView
                    .onGeometryChange(for: CGFloat.self) { proxy in
                        proxy.frame(in: .global).width
                    } action: { newValue in
                        width = newValue * widthScaleFactor
                    }
            }
        }
```

</details>

<details>
<summary>Console Log</summary>

```
*** Assertion failure in -[_UIRegularLegacyScrollerImp setKnobProportion:], _UIScrollerImp.m:347
Invalid parameter not satisfying: !isnan(newKnobProportion)
(
	0   CoreFoundation                      0x000000019ae2ee80 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x000000019a916cd8 objc_exception_throw + 88
	2   Foundation                          0x000000019c03d978 -[NSCalendarDate initWithCoder:] + 0
	3   UIKitCore                           0x00000001cd391410 -[_UIScrollerImp setKnobProportion:] + 296
	4   UIKitCore                           0x00000001cd390c54 __54-[UIScrollView _updateScrollerImpsForcingReplacement:]_block_invoke + 232
	5   UIKitCore                           0x00000001cd38e91c -[UIScrollView _updateScrollerImpsForcingReplacement:] + 1816
	6   UIKitCore                           0x00000001cd3e39cc -[UIScrollView setContentSize:] + 1008
	7   SwiftUI                             0x00000001e9688a94 $s7SwiftUI17HostingScrollViewC13updateContextySbAA0cde6UpdateG0VF + 5244
	8   SwiftUI                             0x00000001e9909344 $s7SwiftUI24UpdatedHostingScrollView33_2827C9FDBC03BB2C61476E3A694FFF5DLLV11updateValueyyF + 5224
	9   SwiftUI                             0x00000001e87444c0 $s14AttributeGraph0A0VyACyxGqd__c5ValueQyd__RszAA12StatefulRuleRd__lufcADSPyqd__GXEfU_ySv_So11AGAttributeatcyXEfU_ySv_AJtcfu_7SwiftUI17HostingScrollViewC_AK07UpdatedijK033_2827C9FDBC03BB2C61476E3A694FFF5DLLVTgm5 + 24
	10  AttributeGraph                      0x00000001cb16cb24 _ZN2AG5Graph11UpdateStack6updateEv + 540
	11  AttributeGraph                      0x00000001cb16d414 _ZN2AG5Graph16update_attributeENS_4data3ptrINS_4NodeEEEj + 424
	12  AttributeGraph                      0x00000001cb17520c _ZN2AG5Graph9value_refENS_11AttributeIDEjPK15AGSwiftMetadataRh + 288
	13  AttributeGraph                      0x00000001cb18cf84 AGGraphGetWeakValue + 388
	14  SwiftUICore                         0x000000023a3d0538 $s7SwiftUI9GraphHostC17updatePreferencesSbyF + 64
	15  SwiftUICore                         0x000000023a7932b0 $s7SwiftUI9ViewGraphC13updateOutputs33_7D9EDEF832940A362646A6E979F296C8LL5asyncySb_tF + 392
	16  SwiftUICore                         0x000000023a788b10 $s7SwiftUI16ViewRendererHostPAAE6render8interval17updateDisplayList15targetTimestampySd_SbAA4TimeVSgtFyyXEfU_yyXEfU0_ + 152
	17  SwiftUICore                         0x000000023a78890c $s7SwiftUI16ViewRendererHostPAAE6render8interval17updateDisplayList15targetTimestampySd_SbAA4TimeVSgtFyyXEfU_ + 708
	18  SwiftUICore                         0x000000023a785ed0 $s7SwiftUI16ViewRendererHostPAAE6render8interval17updateDisplayList15targetTimestampySd_SbAA4TimeVSgtF + 556
	19  SwiftUI                             0x00000001e8fb8ca0 $s7SwiftUI17UIHostingViewBaseC14layoutSubviewsyyF + 808
	20  SwiftUI                             0x00000001e9752c04 $s7SwiftUI14_UIHostingViewC14layoutSubviewsyyF + 120
	21  SwiftUI                             0x00000001e9752c40 $s7SwiftUI14_UIHostingViewC14layoutSubviewsyyFTo + 36
	22  UIKitCore                           0x00000001cd36d950 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1856
	23  QuartzCore                          0x00000001a390575c _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 516
	24  UIKitCore                           0x00000001cd36ce30 -[UIView(Hierarchy) layoutBelowIfNeeded] + 304
	25  SwiftUI                             0x00000001e902ef38 $s7SwiftUI35UICollectionViewListCoordinatorBaseC6update_2to11transaction11performDiffySo0cD0C_xAA11TransactionVSbtF0K7UpdatesL_yyAA010Collectionde10DataSourceG0RzAA16SelectionManagerR_0R5ValueQy_AORtzr0_lFyyXEfU1_TA + 36
	26  SwiftUI                             0x00000001e893832c $s7SwiftUI19SubmitTriggerSourcePAAE14dispatchUpdateyyyyXEFyyXEfU_ + 28
	27  SwiftUI                             0x00000001e8c9505c $sIeg_IyB_TR + 28
	28  UIKitCore                           0x00000001cd342918 +[UIView(Animation) performWithoutAnimation:] + 76
	29  SwiftUI                             0x00000001e9028280 $s7SwiftUI35UICollectionViewListCoordinatorBaseC6update_2to11transaction11performDiffySo0cD0C_xAA11TransactionVSbtF0K7UpdatesL_yyAA010Collectionde10DataSourceG0RzAA16SelectionManagerR_0R5ValueQy_AORtzr0_lF + 1780
	30  SwiftUI                             0x00000001e893832c $s7SwiftUI19SubmitTriggerSourcePAAE14dispatchUpdateyyyyXEFyyXEfU_ + 28
	31  SwiftUI                             0x00000001e8c9505c $sIeg_IyB_TR + 28
	32  UIKitCore                           0x00000001cd342918 +[UIView(Animation) performWithoutAnimation:] + 76
	33  SwiftUI                             0x00000001e902abc4 $s7SwiftUI35UICollectionViewListCoordinatorBaseC6update_2to11transaction11performDiffySo0cD0C_xAA11TransactionVSbtFyycfU_ + 348
	34  SwiftUICore                         0x0000000239f1e444 $sIeg_ytIegr_TRTA + 28
	35  SwiftUICore                         0x000000023a1ddf6c $s7SwiftUI6UpdateO15dispatchActionsyyFZ + 1236
	36  SwiftUICore                         0x000000023a1dd450 $s7SwiftUI6UpdateO3endyyFZ + 212
	37  SwiftUICore                         0x000000023a7076e0 $sSo9NSRunLoopC7SwiftUIE11addObserveryyyycFZySo05CFRunbF3RefaSg_So0gB8ActivityVSvSgtcfU_Tf4ddd_n + 176
	38  CoreFoundation                      0x000000019adbbbd8 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 36
	39  CoreFoundation                      0x000000019adbbac4 __CFRunLoopDoObservers + 552
	40  CoreFoundation                      0x000000019adba770 CFRunLoopRunSpecific + 664
	41  HIToolbox                           0x00000001a6312530 RunCurrentEventLoopInMode + 292
	42  HIToolbox                           0x00000001a6318348 ReceiveNextEventCommon + 676
	43  HIToolbox                           0x00000001a6318508 _BlockUntilNextEventMatchingListInModeWithFilter + 76
	44  AppKit                              0x000000019e925034 _DPSNextEvent + 660
	45  AppKit                              0x000000019f2892d4 -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 688
	46  AppKit                              0x000000019e918060 -[NSApplication run] + 480
	47  AppKit                              0x000000019e8ee854 NSApplicationMain + 888
	48  AppKit                              0x000000019eb3cbcc +[NSWindow _savedFrameFromString:] + 0
	49  UIKitMacHelper                      0x00000001b5828940 UINSApplicationMain + 972
	50  UIKitCore                           0x00000001cd2e5654 UIApplicationMain + 148
	51  SwiftUI                             0x00000001e904cc14 $s7SwiftUI17KitRendererCommon33_ACC2C5639A7D76F611E170E831FCA491LLys5NeverOyXlXpFAESpySpys4Int8VGSgGXEfU_ + 168
	52  SwiftUI                             0x00000001e904c918 $s7SwiftUI6runAppys5NeverOxAA0D0RzlF + 100
	53  SwiftUI                             0x00000001e8cf25c8 $s7SwiftUI3AppPAAE4mainyyFZ + 224
	54  Toolkit Examples.debug.dylib        0x0000000102589720 $s16Toolkit_Examples0B3AppV5$mainyyFZ + 152
	55  Toolkit Examples.debug.dylib        0x0000000102589800 __debug_main_executable_dylib_entry_point + 12
	56  dyld                                0x000000019a954274 start + 2840
)
FAULT: NSInternalInconsistencyException: Invalid parameter not satisfying: !isnan(newKnobProportion); {
    NSAssertFile = "_UIScrollerImp.m";
    NSAssertLine = 347;
}
```

</details>

Reply:
That's strange. All well.